### ISSUE TITLE
StackOverflowError in delegating TypeAdapterFactory

### ISSUE THREAD

**Issue by daniel-huss** [google/gson#675]
GSON Version: 2.3.1
Description: ```gson.getDelegateAdapter(this, type)``` causes inifinite recursion when used with a @JsonAdapter annotated type.

Minimal example:
```java
package test.gson;

import com.google.gson.Gson;
import com.google.gson.TypeAdapter;
import com.google.gson.TypeAdapterFactory;
import com.google.gson.annotations.JsonAdapter;
import com.google.gson.reflect.TypeToken;

/* @formatter:off */
public class GsonStackOverflowError {

    public static void main(final String[] args) {
        new Gson().fromJson("{}", A.class);
    }

    @JsonAdapter(Adapter.class)
    public static class A {}

    public static class Adapter implements TypeAdapterFactory {
        @Override
        public <T> TypeAdapter<T> create(final Gson gson, final TypeToken<T> type)
        {
            gson.getDelegateAdapter(this, type);
            throw new IllegalStateException("Not reproducible");
        }
    }
}
```

**Comment by daniel-huss** [comment]
GSON 2.3 throws IllegalArgumentException instead, see http://stackoverflow.com/questions/29445165/

**Comment by swankjesse** [comment]
Yup, `getDelegateAdapter()` doesn't work with `@JsonAdapter`. Needs to be fixed.

**Comment by manishgo** [comment]
Any updates on this issue?

**Comment by kevinmost** [comment]
I can confirm that the issue is resolved as of Gson 2.7. Should this issue be closed?

**Comment by JakeWharton** [comment]
Thanks for following up
